packages/js-drive/lib/identity/masternode/handleUpdatedPubKeyOperatorFactory.js

```diff
@@ -0,0 +1,112 @@
+const { hash } = require('@dashevo/dpp/lib/util/hash');
+const Identifier = require('@dashevo/dpp/lib/identifier/Identifier');
+const IdentityPublicKey = require('@dashevo/dpp/lib/identity/IdentityPublicKey');
+const Transaction = require('@dashevo/dashcore-lib/lib/transaction');
+
+/**
+ *
+ * @param {DashPlatformProtocol} transactionalDpp
+ * @param {DriveStateRepository|CachedStateRepositoryDecorator} stateRepository
+ * @param {createMasternodeIdentity} createMasternodeIdentity
+ * @param {DataContractStoreRepository} dataContractRepository
+ * @param {Identifier} masternodeRewardSharesContractId
+ * @return {handleUpdatedPubKeyOperator}
+ */
+function handleUpdatedPubKeyOperatorFactory(
+  transactionalDpp,
+  stateRepository,
+  createMasternodeIdentity,
+  dataContractRepository,
+  masternodeRewardSharesContractId,
+) {
+  /**
+   * @typedef handleUpdatedPubKeyOperator
+   * @param {SimplifiedMNListEntry} masternodeEntry
+   * @param {SimplifiedMNListEntry} previousMasternodeEntry
+   * @return {Promise<{
+   *            create: Document[],
+   *            delete: Document[],
+   * }>}
+   */
+  async function handleUpdatedPubKeyOperator(masternodeEntry, previousMasternodeEntry) {
+    const rawTransaction = await stateRepository
+      .fetchTransaction(masternodeEntry.proRegTxHash);
+
+    const { extraPayload: proRegTxPayload } = new Transaction(rawTransaction.data);
+
+    const documentsToCreate = [];
+    let documentsToDelete = [];
+
+    if (proRegTxPayload.operatorReward > 0) {
+      const proRegTxHash = Buffer.from(masternodeEntry.proRegTxHash, 'hex');
+
+      const operatorIdentityHash = hash(
+        Buffer.concat([
+          proRegTxHash,
+          Buffer.from(masternodeEntry.pubKeyOperator, 'hex'),
```

Comment:
I would use `proRegTxPayload.pubKeyOperator`. Make sure you not reading hex to Buffer many times for the same thing please.